sphinx/themes/basic/static/searchtools.js

```diff
@@ -428,7 +428,11 @@ var Search = {
       var valid = true;
 
       // check if all requirements are matched
-      if (fileMap[file].length != searchterms.length)
+      var filteredTerms = // as search terms with length < 3 are discarded: ignore
+        searchterms.filter(function(term){return term.length > 2}).length
+      var allRequirementsMatched = fileMap[file].length != searchterms.length &&
```

Comment:
IMO, this variable is not needed. I prefer to implement this as simple if-condition. What do you think?

Reply:
I agree. It's more concise without this assignment. I will work in the feedback and merge when the CI is green.